library/core/src/num/mod.rs

```diff
@@ -51,6 +51,10 @@ mod overflow_panic;
 mod saturating;
 mod wrapping;
 
+/// 100% perma-unstable
+#[doc(hidden)]
+pub mod niche_types;
```

Comment:
It's a shame there isn't a way to say "this is pub only for alloc/std". But I guess having it be `unstable` is good enough.